test/test_transforms_v2.py

```diff
@@ -5324,6 +5324,10 @@ def was_applied(output, inpt):
     for input, output in zip(other_inputs, other_outputs):
         assert transform.was_applied(output, input)
 
+    def test_assertions_p_and_max_transforms(self):
+        with pytest.raises(ValueError, match="Only one of `p` and `max_transforms` should be specified."):
+            transforms.RandomChoice([transforms.Pad(2), transforms.RandomCrop(28)], p=[1], max_transforms=3)
```

Comment:
Perhaps this is due to the fact that this file has change a lot in the past, but this test is now part of  `test_pure_tensor_heuristic()` and it shouldn't be there. We have a `def test_random_choice()` somewhere, maybe we should make it a class and put those tests together.
Also, it'd be nice to test the behaviour of the new `max_transforms` parameter